### PR TITLE
Deleting cells should select the next cell

### LINKED ISSUE #429: Deleting cell selects cell above instead of the next cell
The title says it all. If I delete a cell with 'd d', instead of the next cell, the previous cell is selected (as opposed to the classic notebook)

Windows 7, JupyterLab 0.1.1 (latest from conda-forge)

### PR DESCRIPTION
**Pull request by blink1073**
Fixes #429.

### CODE REVIEW

**Comment by sccolbert** [comment]
Does this handle deleting the last cell, in which case the previous cell should be selected? What about deleting the only cell?

**Comment by blink1073** [comment]
Good point, more tests required!

**Comment by blink1073** [comment]
We were testing the second condition already, and now the first.

**Comment by ellisonbg** [comment]
Deleting the last cell should always trigger a new cell to be inserted. This constraint (that a notebook should always have at least 1 cell should probably be enforced at the model level so that other plugins that modify the model directly can't the model into a state that doesn't make sense.

**Comment by sccolbert** [comment]
I still don't think this index logic is correct. Consider the case where we have the following cells:

`[0, 1, 2, 3]`

Assume that cells `2` and `3` are selected and being deleted. In that case, the loop computes `index` as `3`, from which `length - 1 = 1` is subtracted. This results in an `activeIndex` of `2`, which is out of range for the remaining cells of `[0, 1]`.

**Comment by sccolbert** [comment]
@ellisonbg What you describe is the current behavior already: https://github.com/blink1073/jupyterlab/blob/f913c38325f9f1d07a3c3f81dda8894af558eb5a/src/notebook/notebook/actions.ts#L199

**Comment by blink1073** [comment]
I'll look at this in the morning, I'm down a relativePath rabbit hole at the moment...

**Comment by sccolbert** [comment]
No rush.

**Comment by blink1073** [comment]
I agree that deleting the *only* cell should trigger the model to add a new code cell.  Deleting the *last* cell in the "classic" notebook activates the previous cell.  @sccolbert, to address your point about the indexing, the `activeCellIndex` is already clamped to the bounds of available cells.

**Comment by blink1073** [comment]
Updated.

File: src/notebook/notebook/actions.ts [hunk 1]
```diff
@@ -195,18 +193,12 @@ namespace NotebookActions {
...
-    }
+    // The model will add a new code cell if there are no
+    // remaining cells.
     model.cells.endCompoundOperation();
 
-    // Activate the previous cell.
-    if (index === -1) {
-      index = 0;
-    }
+    // Select the cell *after* the last selected.
```
Comment (jasongrout) [inline src/notebook/notebook/actions.ts]: Can you add a comment that clamping the index on setting it ensures that if the last cell was deleted, the new last cell is selected? That part is a bit magical, and it makes sure we are clear for the future on what the behavior should be.

**Comment by jasongrout** [comment]
+1, modulo adding a short comment explaining that clamping makes the 'delete last cell' case work.

**Comment by blink1073** [comment]
Ready.  There was an interesting bug on Travis due to binding signals to the window object in the activity monitor tests, now fixed.

**Comment by jasongrout** [comment]
Great! :shipit:

**Pull request merged**